src/sentry/ingest/consumer/processors.py

```diff
@@ -229,9 +229,19 @@ def process_event(
         except Exception:
             pass
 
-        project.set_cached_field_value(
-            "organization", Organization.objects.get_from_cache(id=project.organization_id)
-        )
+        try:
+            project.set_cached_field_value(
+                "organization", Organization.objects.get_from_cache(id=project.organization_id)
+            )
+        except Organization.DoesNotExist:
+            logger.warning(
+                "Organization does not exist",
+                extra={
+                    "project_id": project_id,
+                    "organization_id": project.organization_id,
+                },
+            )
+            return
```

Comment:
I am not sure we need to return here. To preserve the old behavior I would just go ahead without pre-populating the org.